py-polars/src/conversion.rs

```diff
@@ -839,7 +839,9 @@ impl<'s> FromPyObject<'s> for Wrap<AnyValue<'s>> {
                             get_bool
                             // TODO: this heap allocs on failure
                         } else if ob.extract::<i64>().is_ok() {
-                            get_int
+                            get_i64
+                        } else if ob.extract::<u64>().is_ok() {
+                            get_u64
```

Comment:
I think this should remain `get_int` and `get_int` should remain unchanged as it will correct for overflow.

The extra branch is correct. That was the reason we skipped `i64` and got into the float branch.

However, we do need the generic `get_int` function as both small integers and large integers will have the same place in the lookup table.